BlackMarket/app/SignUp/SignUpCardView.swift

```diff
@@ -0,0 +1,211 @@
+//
+//  SignUpCardView.swift
+//  BlackMarket
+//
+//  Created by Tarek Radovan on 06/12/2022.
+//
+
+import SwiftUI
+import ComposableArchitecture
+
+struct SignUpCardDomain: ReducerProtocol {
+  
+  struct State: Equatable {
+    static func == (lhs: SignUpCardDomain.State, rhs: SignUpCardDomain.State) -> Bool {
+      return true
+    }
+   
+    var nameState = BMTextFieldDomain.State(
+      validations: [.nonEmpty],
+      placeholder: LocalizedString.SignUpCardView.nameTextfieldPlaceholder,
+      errorMessage: LocalizedString.SignUpCardView.nameTextfieldError,
+      title: LocalizedString.SignUpCardView.nameTextfieldTitle
+    )
+    var emailState = BMTextFieldDomain.State(
+      validations: [.email, .nonEmpty],
+      placeholder: LocalizedString.SignUpCardView.emailTextfieldPlaceholder,
+      errorMessage: LocalizedString.SignUpCardView.emailTextfieldError,
+      title: LocalizedString.SignUpCardView.emailTextfieldTitle
+    )
+    var passwordState = BMTextFieldDomain.State(
+      validations: [
+        .custom(isValid: { value in
+          value.count >= Constants.Value.minimumPasswordLength
+        })
+      ],
+      placeholder: LocalizedString.SignUpCardView.passwordTextfieldPlaceholder,
+      errorMessage: LocalizedString.SignUpCardView.passwordTextfieldError,
+      title: LocalizedString.SignUpCardView.passwordTextfieldTitle,
+      isSecure: true
+    )
+    
+    var dataPolicyLink: AttributedString {
+      LocalizedString.SignUpCardView.dataPolicy.transformToLink(
+        withURL: Constants.Path.dataPolicy
+      ) ?? ""
+    }
+    var cookiesPolicyLink: AttributedString {
+      LocalizedString.SignUpCardView.cookiesPolicy.transformToLink(
+        withURL: Constants.Path.cookiesPolicy
+      ) ?? ""
+    }
+  }
+  
+  enum Action {
+    case nameChanged(BMTextFieldDomain.Action)
+    case emailChanged(BMTextFieldDomain.Action)
+    case passwordChanged(BMTextFieldDomain.Action)
+    case registerUser
+    case showPolicy
+    case presentLogIn
+    case showCookiesPolicy
+  }
+  
+  var body: some ReducerProtocol<State, Action> {
+    Reduce { state, action in
+      switch action {
+      case .registerUser:
+        return .none
+      case .showPolicy:
+        return .none
+      case .presentLogIn:
+        return .none
+      case .showCookiesPolicy:
+        return .none
+      default:
+        return .none
+      }
```

Comment:
for now this could be just:

```suggestion
.none
```

right?

Reply:
Yeah, for now we can return just .none